skimage/registration/_optical_flow.py

```diff
@@ -0,0 +1,215 @@
+                       indexing='ij')
+
+    dt = 0.5/image0.ndim
+    reg_niter = 2
+    f0 = attachment * tightness
+    f1 = dt / tightness
+    tol *= image0.size
+
+    flow_current = flow_previous = flow0
+
+    g = np.zeros((image0.ndim, ) + image0.shape, dtype=dtype)
+    proj = np.zeros((image0.ndim, image0.ndim, ) + image0.shape,
+                    dtype=dtype)
+
+    s_g = [slice(None), ] * g.ndim
+    s_p = [slice(None), ] * proj.ndim
+    s_d = [slice(None), ] * (proj.ndim-2)
+
+    for _ in range(nwarp):
+        if prefilter:
+            flow_current = ndi.median_filter(flow_current,
+                                             [1]+image0.ndim*[3])
+
+        image1_warp = warp(image1, grid+flow_current, mode='nearest')
+        grad = np.array(np.gradient(image1_warp))
+        NI = (grad*grad).sum(0)
+        NI[NI == 0] = 1
+
+        rho_0 = image1_warp - image0 - (grad*flow_current).sum(0)
+
+        for _ in range(niter):
+
+            # Data term
+
+            rho = rho_0 + (grad*flow_current).sum(0)
+
+            idx = abs(rho) <= f0 * NI
+
+            flow_auxiliary = flow_current
+
+            flow_auxiliary[:, idx] -= rho[idx]*grad[:, idx]/NI[idx]
+
+            idx = ~idx
+            srho = f0 * np.sign(rho[idx])
+            flow_auxiliary[:, idx] -= srho*grad[:, idx]
+
+            # Regularization term
+            flow_current = flow_auxiliary.copy()
+
+            for idx in range(image0.ndim):
+                s_p[0] = idx
+                for _ in range(reg_niter):
+                    for ax in range(image0.ndim):
+                        s_g[0] = ax
+                        s_g[ax+1] = slice(0, -1)
+                        g[tuple(s_g)] = np.diff(flow_current[idx], axis=ax)
+                        s_g[ax+1] = slice(None)
+
+                    norm = np.sqrt((g ** 2).sum(0))[np.newaxis, ...]
+                    norm *= f1
+                    norm += 1.
+                    proj[idx] -= dt * g
+                    proj[idx] /= norm
+
+                    # d will be the (negative) divergence of proj[idx]
+                    d = -proj[idx].sum(0)
+                    for ax in range(image0.ndim):
+                        s_p[1] = ax
+                        s_p[ax+2] = slice(0, -1)
+                        s_d[ax] = slice(1, None)
+                        d[tuple(s_d)] += proj[tuple(s_p)]
+                        s_p[ax+2] = slice(None)
+                        s_d[ax] = slice(None)
+
+                    flow_current[idx] = flow_auxiliary[idx] + d
+
+        flow_previous -= flow_current  # The difference as stopping criteria
+        if (flow_previous*flow_previous).sum() < tol:
+            break
+
+        flow_previous = flow_current
+
+    return flow_current
+
+
+def optical_flow_tvl1(image0, image1, *, attachment=15, tightness=0.3,
+                      nwarp=5, niter=10, tol=1e-4, prefilter=False,
+                      dtype='float32'):
+    """Coarse to fine optical flow estimator.
```

Comment:
```suggestion
    r"""Coarse to fine optical flow estimator.
```

This is needed because the docstring contains math (`\lambda` etc), which can be ambiguous in non-raw strings.